The PreCommit Java Debezium IO Direct job is flaky

The PreCommit Java Debezium IO Direct is failing over 50% of the time.
Please visit https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java_Debezium_IO_Direct.yml?query=is%3Afailure+branch%3Amaster to see all failed workflow runs.
See also Grafana statistics: http://metrics.beam.apache.org/d/CTYdoxP4z/ga-post-commits-status?orgId=1&viewPanel=3&var-Workflow=PreCommit%20Java%20Debezium%20IO%20Direct

now it fails with

testDebeziumSchemaTransformPostgresRead

```
Caused by: java.lang.NoSuchMethodError: 'void io.debezium.connector.postgresql.proto.PgProto$RowMessage.makeExtensionsImmutable()'
	at io.debezium.connector.postgresql.proto.PgProto$RowMessage.<init>(PgProto.java:3509)
	at io.debezium.connector.postgresql.proto.PgProto$RowMessage.<init>(PgProto.java:3381)
	at io.debezium.connector.postgresql.proto.PgProto$RowMessage$1.parsePartialFrom(PgProto.java:5237)
	at io.debezium.connector.postgresql.proto.PgProto$RowMessage$1.parsePartialFrom(PgProto.java:5231)
	at com.google.protobuf.AbstractParser.parsePartialFrom(AbstractParser.java:135)
	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:168)
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -72,7 +72,7 @@
 ## Breaking Changes
 
 * AWS V1 I/Os have been removed (Java). As part of this, x-lang Python Kinesis I/O has been updated to consume the V2 IO and it also no longer supports setting producer_properties ([#33430](https://github.com/apache/beam/issues/33430)).
-* Upgraded to protobuf 4 (Java) ([#33192](https://github.com/apache/beam/issues/33192)), but forced Debezium IO to use protobuf 3 ([#TODO](https://github.com/apache/beam/issues/TODO) because Debezium clients are not protobuf 4 compatible. This may cause conflicts when using clients which are only compatible with protobuf 4.
+* Upgraded to protobuf 4 (Java) ([#33192](https://github.com/apache/beam/issues/33192)), but forced Debezium IO to use protobuf 3 ([#33541](https://github.com/apache/beam/issues/33541) because Debezium clients are not protobuf 4 compatible. This may cause conflicts when using clients which are only compatible with protobuf 4.
 
 ## Deprecations
 

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -67,10 +67,12 @@
 ## New Features / Improvements
 
 * X feature added (Java/Python) ([#X](https://github.com/apache/beam/issues/X)).
+* Upgraded to protobuf 4 (Java) ([#33192](https://github.com/apache/beam/issues/33192)).
 
 ## Breaking Changes
 
 * AWS V1 I/Os have been removed (Java). As part of this, x-lang Python Kinesis I/O has been updated to consume the V2 IO and it also no longer supports setting producer_properties ([#33430](https://github.com/apache/beam/issues/33430)).
+* Upgraded to protobuf 4 (Java) ([#33192](https://github.com/apache/beam/issues/33192)), but forced Debezium IO to use protobuf 3 ([#TODO](https://github.com/apache/beam/issues/TODO) because Debezium clients are not protobuf 4 compatible. This may cause conflicts when using clients which are only compatible with protobuf 4.
 
 ## Deprecations
 

--- a/sdks/java/io/debezium/build.gradle
+++ b/sdks/java/io/debezium/build.gradle
@@ -89,3 +89,14 @@ task integrationTest(type: Test, dependsOn: processTestResources) {
   useJUnit {
   }
 }
+
+configurations.all (Configuration it) -> {
+  resolutionStrategy {
+    // Force protobuf 3 because debezium is currently incompatible with protobuf 4.
+    // TODO - remove this and upgrade the version of debezium once a proto-4 compatible version is available
+    // https://github.com/apache/beam/pull/33526 does some of this, but was abandoned because it still doesn't
+    // work with protobuf 4.
+    force "com.google.protobuf:protobuf-java:3.25.5"
+    force "com.google.protobuf:protobuf-java-util:3.25.5"
+  }
+}

--- a/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubGrpcClient.java
+++ b/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubGrpcClient.java
@@ -17,7 +17,6 @@
  */
 package org.apache.beam.sdk.io.gcp.pubsub;
 
-import static org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Preconditions.checkArgument;
 import static org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Preconditions.checkState;
 
 import com.google.auth.Credentials;

